### PR TITLE
Libxdp support

### PR DESCRIPTION
**Pull request by cianperill**
- libxdp installation is detected with pkg-config
- libxdp installation is prioritised over libbpf installations when building
- added libxdp info to README.md

### CODE REVIEW

File: internal/bpf/bpfWrapper.c [hunk 1]
```diff
@@ -13,10 +13,13 @@
...
 #include <linux/if_link.h> // for XDP_FLAGS_UPDATE_IF_NOEXIST
 #include <net/if.h>	   // for if_nametoindex
-
+#if __has_include(<xdp/xsk.h>)
+#include <xdp/xsk.h>
+#else
+#include <bpf/xsk.h>
+#endif
 #include "bpfWrapper.h"
 #include "log.h"
```
Comment (maryamtahhan) [inline internal/bpf/bpfWrapper.c:25]: The function

`int Load_bpf_send_xsk_map(char *ifname) `

should be renamed to 

`int Load_bpf_xsk_prog(char *ifname)`

Comment (maryamtahhan) [inline internal/bpf/bpfWrapper.c:25]: Additionally in the cleanup prog below
`bpf_set_link_xdp_fd(if_index, fd, XDP_FLAGS_UPDATE_IF_NOEXIST);` will need to be updated to have an option to use libxdp constructs for cleanup using ```xdp_program__detach()``` as shown in: https://github.com/xdp-project/xdp-tools/blob/master/xdp-loader/xdp-loader.c#L248
OR use xdp-loader utility directly to unload the prog:
```
xdp-loader unload veth0 --all
```

File: Makefile [hunk 2]
```diff
@@ -32,7 +34,10 @@ format:
 buildc:
 	@echo "******     Build BPF     ******"
 	@echo
-	gcc ./internal/bpf/bpfWrapper.c -lbpf -c -o ./internal/bpf/bpfWrapper.o
+    ifeq ($(bpf_lib),)
```
Comment (maryamtahhan) [inline Makefile:37]: so this line is supposed to check if libbpf version is <=0.7 right? the set the bpf_flag to -lbpf
On line 15 we check the libxdp version and set libbpf to it. so is this check always being hit or never being hit as we aren't actually comparing versions to anything?

**Comment by garyloug** [comment]
Going to close this PR as we found it's difficult to install libxdp on all but the most recent non-LTS operating systems. As the CNI runs directly on the host, it means we'd be passing the burden of this library installation onto all users of the plugins.

Will place libxdp support on its own branch for now until it becomes a little more mainstream, so people can build if they want to test with it.

**Pull request closed without merging**